src/sequence.lisp

```diff
@@ -458,6 +458,19 @@
 	  (funcall result-collector (apply function args))
 	  (apply function args)))))
 
+(defun concatenate (result-type &rest sequences)
+  (ecase result-type
```

Comment:
Could we extract this into a `(check-is-sequence X)` function maybe? 

I guess it'll be useful in other places too.

Reply:
Yes, added `check-sequence-type-specifier` in  c55c0ada15ebcef74cf6764de8e6cd387d80e4c3 - what do you think?